.github/workflows/build.yml

```diff
@@ -1,18 +1,51 @@
-name: Build Docker image
+name: Build and test Docker image
 
+# Run builds on main commits and when PRs are opened or updated
 on:
   push:
     branches:
-      - main
+    - main
+  pull_request:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
 
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
     steps:
-    - uses: actions/checkout@v1
-    - name: Publish to Registry
-      uses: elgohr/Publish-Docker-Github-Action@master
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+      with:
+        install: true # make buildx the default
+    - name: Extract metadata for Docker
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+    - name: Log in to the Container registry
+      uses: docker/login-action@v1
+      if: github.ref == 'refs/heads/main'
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Build and test Docker image
+      id: build
+      uses: docker/build-push-action@v2
       with:
-        name: deathandmayhem/jolly-roger
-        username: ${{ secrets.DOCKER_USERNAME }}
-        password: ${{ secrets.DOCKER_PASSWORD }}
+        context: .
+        push: ${{ github.ref == 'refs/heads/main' }} # only push on main builds
```

Comment:
Note that this only runs on `main` so it's possible this will break and we'll need to fix it later.